src/components/KakaoLogin/AdlipSignup.js

```diff
@@ -0,0 +1,236 @@
+import React, { useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import styled from 'styled-components';
+
+function AdlipSignUp({ isHost }) {
+  const history = useHistory();
+  const [user, setUser] = useState({
+    email: '',
+    password: '',
+    passwordCheck: '',
+    nickname: '',
+  });
+
+  function handleChange(e) {
+    const { name, value } = e.target;
+    setUser({ ...user, [name]: value });
+  }
+
+  const validEmail =
+    /^[0-9a-zA-Z]([-_\.]?[0-9a-zA-Z])*@[0-9a-zA-Z]([-_\.]?[0-9a-zA-Z])*\.[a-zA-Z]{2,3}$/i;
+  const validPw =
+    /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[a-zA-Z])(?=.*[!@#\$%\^&\*]).{10,}$/;
+
+  const isValidEmail = user.email.match(validEmail) || user.email === '';
+  const isValidPw = user.password.match(validPw) || user.password === '';
+  const isValidPwCheck =
+    user.passwordCheck === user.password || user.passwordCheck === '';
+  const isValidNickname = user.nickname.length <= 20 || user.nickname === '';
+
+  function signup({ isHost }) {
+    if (isHost) {
+      fetch('user/signup', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: {
+          email: user.email,
+          password: user.password,
+          nickname: user.nickname,
+          status: 'host',
+          socialPlatform: 'adlip',
+        },
```

Comment:
const hostBody = {
          email: user.email,
          password: user.password,
          nickname: user.nickname,
          status: 'host',
          socialPlatform: 'adlip',
        },

const guestBody = {}

```suggestion
        body: isHost ?  hostBody : guestBody
```